contracts/modules/utils/RequireUtils.sol

```diff
@@ -61,11 +77,100 @@ contract RequireUtils {
             )
           )
         )
-      ) == _wallet, "RequireUtils#requireConfig: UNEXPECTED_COUNTERFACTUAL_IMAGE_HASH");
+      ) == _wallet, "RequireUtils#publishConfig: UNEXPECTED_COUNTERFACTUAL_IMAGE_HASH");
     }
 
     // Emit event for easy config retrieval
     emit RequiredConfig(_wallet, imageHash, _threshold, abi.encode(_members));
+
+    if (_index) {
+      // Register last event for given wallet
+      lastWalletUpdate[_wallet] = block.number;
+    }
+  }
+
+  function publishSigners(
+    address _wallet,
+    bytes32 _hash,
```

Comment:
Can someone take a hash that I already signed with my key `K`, then sign it with they keys 'Z' and publish valid signers? If so then one can still poison the logs with arbitrary keys.

Reply:
Yes, the fix for that issue is #106

Reply:
Now that I think more about this, the attacker could use `publishSigners` and poison the index using wallets that don't follow the same implementation as the sequence wallet for `imageHash`, they could create an unlimited amount of wallets that return the same `imageHash`. I need to think a little bit more how to mitigate it.

Reply:
the only reason this method exists is to store the initial config for data availability reasons -- right?

so why don't we rename it to publishInitialSigners, and only allow this value to ever be set once by a wallet. Once set, it cannot be overwritten. That will prevent the poisoning, but of course if someone somehow writes the wrong initial signer config, then they're stuck. Perhaps this percolates some ideas or maybe risk doesnt exist for uploading the wrong init config..? 

or, another idea is we compute the "hash" in the smart contract from the signers, so the signer payload always matches out imageHash encoding